test/fixtures/section-heading-first-fail.html

```diff
@@ -0,0 +1,3 @@
+<section>
+    <p>Hello, world.</p>
+</section>
```

Comment:
The test tries to guarantee that a heading is the first immediate child of a section. Any heading (h1-h6) would work, so I can't think of an invalid heading. Can you clarify?